addons/account/models/account_reconcile_model.py

```diff
@@ -107,6 +107,23 @@ class AccountReconcileModel(models.Model):
     second_analytic_account_id = fields.Many2one('account.analytic.account', string='Second Analytic Account', ondelete='set null')
     second_analytic_tag_ids = fields.Many2many('account.analytic.tag', string='Second Analytic Tags',
                                                relation='account_reconcile_model_second_analytic_tag_rel')
+    
+    number_entries = fields.Integer(string='Number of entries related to this model', compute='_compute_number_entries')
+
+    @api.multi
+    def action_reconcile_stat(self):
+        action = self.env.ref('account.action_move_journal_line').read()[0]
+        action.update({
+            'context': {'search_default_reconcile_model_id': self.name},
+            'help': """<p class="o_view_nocontent_empty_folder">{}</p>""".format(_('No move from this reconciliation model')),
+        })
+        return action
+        
+    @api.multi
+    def _compute_number_entries(self):
+        for model in self:
```

Comment:
It's done, but is there a big difference in terms of speed ? I find it harder to read.

Reply:
it does make a huge difference, indeed... when it's done the right way (which is not yet the case here) ^^

the idea is to get the count for each record in self, in a single query instead of a loop then X queries per record. So you need to move the read_group() call outside of the loop to make it efficient

Reply:
Yes but as far as I know, it is not possible to do a read_group on a 2many field...

Reply:
I also thought of something like this:
```python
    @api.multi
    def _compute_number_entries(self):
        dict = {}
        for move in self.env['account.move'].search([]):
            if move.reconcile_model_id not in dict:
                move.reconcile_model_id = 1
            else:
                move.reconcile_model_id += 1
        for model in self:
            model.number_entries = dict.get(model, 0)
```

But isn't it worse ? We are fetching all the rows of account.move in python instead of doing it in sql.

Reply:
BTW, `self` in this case is always of length 1

Reply:
self.env['account.move.line'].read_group([('reconcile_model_id', 'in', self.ids)], ['reconcile_model_ids'], 'reconcile_model_id')

Reply:
?
and, no, you don't have any certitude that len(self) == 1

Reply:
Ok it's done. Why on earth do I still read the doc?

Reply:
haha